pdp/request_test.go

```diff
@@ -0,0 +1,2093 @@
+package pdp
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"net"
+	"reflect"
+	"strconv"
+	"testing"
+
+	"github.com/infobloxopen/go-trees/domain"
+	"github.com/infobloxopen/go-trees/domaintree"
+	"github.com/infobloxopen/go-trees/iptree"
+	"github.com/infobloxopen/go-trees/strtree"
+)
+
+var (
+	testWireRequest = append([]byte{1, 0}, testWireAttributes...)
+
+	testRequestAssignments = []AttributeAssignment{
+		MakeStringAssignment("string", "test"),
+		MakeBooleanAssignment("boolean", true),
+		MakeIntegerAssignment("integer", 9223372036854775807),
+	}
+)
+
+func TestRequestWireTypesTotal(t *testing.T) {
+	if requestWireTypesTotal != len(requestWireTypeNames) {
+		t.Errorf("Expected number of wire type names %d to be equal to total number of wire types %d",
+			len(requestWireTypeNames), requestWireTypesTotal)
+	}
+}
+
+func TestMarshalRequestAssignments(t *testing.T) {
+	var b [44]byte
+	n, err := MarshalRequestAssignments(b[:], testRequestAssignments)
+	assertRequestBytesBuffer(t, "MarshalRequestAssignments", err, b[:], n, testWireRequest...)
+
+	n, err = MarshalRequestAssignments([]byte{}, testRequestAssignments)
+	assertRequestBufferOverflow(t, "MarshalRequestAssignments", err, n)
+
+	n, err = MarshalRequestAssignments(b[:2], testRequestAssignments)
+	if err == nil {
+		t.Errorf("expected no data put to small buffer but got %d", n)
+	} else if _, ok := err.(*requestBufferOverflowError); !ok {
+		t.Errorf("expected *requestBufferOverflowError but got %T (%s)", err, err)
+	}
```

Comment:
It looks to me `assertRequestBufferOverflow()` used in L42 does exactly what L45-L49 does. Is there any reason why we don't reuse `assertRequestBufferOverflow()` here?

There seems to be other occurrences following too.

Reply:
Nice catch!